zone/zoning.cpp

```diff
@@ -832,6 +832,11 @@ void Client::ZonePC(uint32 zoneID, uint32 instance_id, float x, float y, float z
 
 		zone_mode = zm;
 		if (zm == ZoneToBindPoint) {
+			if (!CanEnterZone(pZoneName)) {
```

Comment:
`instance_id` should be included here

Reply:
`CanEnterZone` uses version. Do we mind making a database call to get version from instance ID?

Reply:
It should be available in the `zd` object earlier in `ZonePC` 

```cpp
	auto zd = GetZoneVersionWithFallback(zoneID, zone->GetInstanceVersion());
	if (zd) {
		pZoneName = strcpy(new char[zd->long_name.length() + 1], zd->long_name.c_str());
	}
```

Reply:
That doesn't use `instance_id` to find the version unfortunately.

Reply:
Will have to change to something like this.
```cpp
const uint8 version = database.GetInstanceVersion(instance_id);

auto zd = GetZoneVersionWithFallback(zoneID, version);
if (zd) {
	pZoneName = strcpy(new char[zd->long_name.length() + 1], zd->long_name.c_str());
}```

Reply:
Reading closer, `zd` version should be fetched **with** `instance_id` that's passed in, not the current zone objects instance_id

We will likely need to do a query regardless in order to make that lookup truly accurate before doing the correct lookup. It can then be used in both places.

We can't avoid a query in this case - can add lookup caching later